pallets/nfts/src/lib.rs

```diff
@@ -164,6 +165,19 @@ decl_module! {
 
             Self::deposit_event(RawEvent::Sealed(id));
         }
+
+        /// Remove an NFT from the storage. This operation is irreversible which means
+        /// once the NFT is removed (burned) from the storage there is no way to
+        /// get it back.
+        /// Must be called by the owner of the NFT.
+        #[weight = T::WeightInfo::burn()]
+        fn burn(origin, id: T::NFTId) {
+            let who = ensure_signed(origin)?;
+            let data = Data::<T>::get(id);
+
+            ensure!(data.owner == who, Error::<T>::NotOwner);
+            let _result = <Self as NFTs>::burn(id);
```

Comment:
change to something more like  `<Self as NFTs>::burn(id).expect("burn always returns Ok(); qed")`

Reply:
Will do :)

Reply:
It should now be as proposed.